src/async_connection.rs

```diff
@@ -0,0 +1,543 @@
+use std::io;
```

Comment:
Yeah, this was a prototype I wrote while trying to figure out how tokio worked; I ended up splitting it out as it got more complex and hadn't taken the time to clean up the older cruft in the branch yet.